fix: Fix test bug for InfluxRPC `<unknown field> != <value>`

crepererum opened the pull request:
Fixes #4786.

crepererum commented on predicate/src/rpc_predicate/ignore_unknown.rs:
```diff
@@ -0,0 +1,42 @@
+    fn mutate(&mut self, expr: Expr) -> DataFusionResult<Expr> {
+        if let Expr::BinaryExpr {
+            left,
+            op: Operator::NotEq,
```
Should we only ignore `!=` or should we also ignore other operators? (esp. "regex NOT match")

alamb reviewed (commented):
This is a neat idea @crepererum  -- the code is quite beautiful. However, TLDR is I don't think the `list_tag_columns_predicate_negative_nonexistent_column`  test is quite right.

Specifically, rather than 
```rust
.with_expr(col("host").not_eq(lit("server01")))
``` 

it should be something like

```rust
.with_expr(make_empty_tag_ref_expr("host").not_eq(lit("server01")))
``` 

Using `make_empty_tag_ref_expr` defined in https://github.com/influxdata/influxdb_iox/blob/master/query_tests/src/influxrpc/read_filter.rs#L805

I suspect if you adjust the test the code changes will not be needed. Specifically, the way a predicate like `host != 'server1'` from influxRPC is supposed to work is:

# Step 1
The predicate comes encoded as `TagRef('host') != 'server1'` on the storage grpc proto

# Step 2
`TagRef('host')` is transformed to `CASE WHEN host IS NULL THEN '' ELSE host END` resulting in the predicate like

```sql
CASE WHEN host IS NULL THEN '' ELSE host END != 'server1'
```

This happens in https://github.com/influxdata/influxdb_iox/blob/master/service_grpc_influxrpc/src/expr.rs#L511


# Step 3
The predicate rewrite code replaces references to any non existent columns with `NULL`. So in this case, the predicate will be rewritten to

```sql
CASE WHEN NULL IS NULL THEN '' ELSE host END != 'server1'
```
-->
```sql
 '' != 'server1'
```

-->
```sql
true
```

crepererum commented:
@alamb You're right, then the test passes. So I've removed the new code again and just adjusted the test. I think that still counts as "closing the issue", because it turned out to be a test bug.

alamb commented:
>  think that still counts as "closing the issue", because it turned out to be a test bug.

I agree -- thank you

alamb commented on query_tests/src/influxrpc/tag_keys.rs:
```diff
@@ -7,6 +7,8 @@ use iox_query::{
 };
 use predicate::{rpc_predicate::InfluxRpcPredicate, Predicate};
 
+use super::util::make_empty_tag_ref_expr;
```
👍

alamb reviewed (approved):
Thanks @crepererum . FYI @carols10cents
